alphacsc/utils/coordinate_descent.py

```diff
@@ -56,15 +56,15 @@ def _coordinate_descent_idx(Xi, D, constants, reg, z0=None, max_iter=1000,
 
     if n_seg == 'auto':
         if strategy == 'greedy':
-            n_seg = max(n_times_valid // (2 * n_times_atom), 1)
+            n_seg = max(n_times_valid // (2 * n_times_atom - 1), 1)
         elif strategy in ('random', 'cyclic'):
             n_seg = 1
             n_coordinates = n_times_valid * n_atoms
 
     rng = check_random_state(random_state)
 
     max_iter *= n_seg
-    n_times_seg = n_times_valid // n_seg + 1
+    n_times_seg = n_times_valid // n_seg
 
```

Comment:
i think the test are failing because with this values, you don't cover all the coordinate with the segments.
You need to have enough segment so that `n_seg * n_times_seg > n_times_valid`.

Reply:
Ok. I understand. I had tried to make segment counts equal with dicodile. I did not pay attention to that. I've reverted and fixed on dicodile side (If I understand correct). 

I want to add a test to see that dicodile and alphacsc gives the same values after each iteration.